datasets/loader.py

```diff
@@ -63,30 +121,160 @@ def a9a(dataset_dir=None):
+    for data, name in zip((X_train, X_test, y_train, y_test),
+                          ('x_train', 'x_test', 'y_train', 'y_test')):
+        filename = f'{dataset_name}_{name}.npy'
+        np.save(os.path.join(dataset_dir, filename), data)
+    logging.info(f'dataset {dataset_name} is ready.')
+    return True
+
+
+def airline_ohe(dataset_dir: Path) -> bool:
     """
-    Author: Danil Prokhorov.
-    libSVM,AAD group
-    Cite: Danil Prokhorov. IJCNN 2001 neural network competition.
-    Slide presentation in IJCNN'01,
-    Ford Research Laboratory, 2001. http://www.geocities.com/ijcnn/nnc_ijcnn01.pdf.
+    Dataset from szilard benchmarks: https://github.com/szilard/GBM-perf
+    TaskType:binclass
+    NumberOfFeatures:700
+    NumberOfInstances:10100000
+    """
+    dataset_name = 'airline-ohe'
+    os.makedirs(dataset_dir, exist_ok=True)
+
+    url_train = 'https://s3.amazonaws.com/benchm-ml--main/train-10m.csv'
+    url_test = 'https://s3.amazonaws.com/benchm-ml--main/test.csv'
+    local_url_train = os.path.join(dataset_dir, os.path.basename(url_train))
+    local_url_test = os.path.join(dataset_dir, os.path.basename(url_test))
+    if not os.path.isfile(local_url_train):
+        logging.info(f'Started loading {dataset_name}')
+        _retrieve(url_train, local_url_train)
+    if not os.path.isfile(local_url_test):
+        logging.info(f'Started loading {dataset_name}')
+        _retrieve(url_test, local_url_test)
+    logging.info(f'{dataset_name} is loaded, started parsing...')
+
+    sets = []
+    labels = []
+
+    categorical_names = ["Month", "DayofMonth",
+                         "DayOfWeek", "UniqueCarrier", "Origin", "Dest"]
+
+    for local_url in [local_url_train, local_url_train]:
+        df = pd.read_csv(local_url, nrows=1000000
+                         if local_url.endswith('train-10m.csv') else None)
+        X = df.drop('dep_delayed_15min', 1)
+        y = df["dep_delayed_15min"]
+
+        y_num = np.where(y == "Y", 1, 0)
+
+        sets.append(X)
+        labels.append(y_num)
+
+    n_samples_train = sets[0].shape[0]
+
+    X_final: Any = pd.concat(sets)
+    X_final = pd.get_dummies(X_final, columns=categorical_names)
+    sets = [X_final[:n_samples_train], X_final[n_samples_train:]]
+
+    for data, name in zip((sets[0], sets[1], labels[0], labels[1]),
+                          ('x_train', 'x_test', 'y_train', 'y_test')):
+        filename = f'{dataset_name}_{name}.npy'
+        np.save(os.path.join(dataset_dir, filename), data)
+    logging.info(f'dataset {dataset_name} is ready.')
+    return True
+
+
+def bosch(dataset_dir: Path) -> bool:
+    dataset_name = 'bosch'
+    os.makedirs(dataset_dir, exist_ok=True)
+
+    filename = "train_numeric.csv.zip"
+    local_url = os.path.join(dataset_dir, filename)
+
+    if not os.path.isfile(local_url):
+        logging.info(f'Started loading {dataset_name}')
+        os.system(
+            "kaggle competitions download -c bosch-production-line-performance -f " +

```

Comment:
We have python kaggle module (pip install kaggle)

Reply:
Need to think about error handling if we can't find kaggle API and don't interrupt overall benchmarks execution

Reply:
I agree about the need in error handling, but I'm not sure if we should put `pip install kaggle` here. This command will not help any user anyway, because we also need to configure kaggle module before work. And configuration data is not public, is it?

Reply:
https://github.com/RukhovichIV/scikit-learn_bench/blob/xgb-nvidia-datasets/datasets/load_datasets.py#L62
And we already have such handling btw